src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/ConnectHelper.cs

```diff
@@ -39,6 +39,38 @@ public CertificateCallbackMapper(Func<HttpRequestMessage, X509Certificate2, X509
         }
 
         public static async ValueTask<(Socket, Stream)> ConnectAsync(string host, int port, CancellationToken cancellationToken)
+        {
+            CancellationTokenSource cancelIPv6 = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
+            Task<(Socket, Stream)> tryConnectAsyncIPv6 = ConnectAsyncInternal(host, port, AddressFamily.InterNetworkV6, cancelIPv6.Token);
+
+            if (await Task.WhenAny(tryConnectAsyncIPv6, Task.Delay(200)).ConfigureAwait(false) == tryConnectAsyncIPv6 && tryConnectAsyncIPv6.IsCompletedSuccessfully)
+            {
+                return await tryConnectAsyncIPv6.ConfigureAwait(false);
+            }
+
+            CancellationTokenSource cancelIPv4 = CancellationTokenSource.CreateLinkedTokenSource(cancellationToken);
```

Comment:
Can we not use the same cancelIPv6 token source that was created earlier?  If the operation has already completed, it won't hurt anything to cancel the token passed to it, so it seems like we should be able to just have a single token that we pass to both the IPv4 and IPv6 task and then cancel at the end to cancel any remaining items.

Reply:
Yes, I think you're right about this. I was concerned about cancelling the completed operation, but that isn't really an issue. I'll address this along with the other cancellation issues brought up here.